File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        sort(points.begin(), points.end(), [](auto& x, auto& y) {
            return x[0] * x[0] + x[1] * x[1] < y[0] * y[0] + y[1] * y[1];
        });
        vector<vector<int>> ans(k);
        for (int i=0; i<k; i++) {
            ans[i] = points[i];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/sum-of-two-integers/submission-0.java
class Solution {
    public int getSum(int a, int b) {
        int ans = 0;
        int carry = 0;
        for (int i=0; i<32; i++) {
            int aSet = (a >> i) & 1;
            int bSet = (b >> i) & 1;
            int curr = aSet ^ bSet ^ carry;
            carry = (aSet & bSet) | (aSet & carry) | (bSet & carry);

            if (curr != 0) {
                ans |= (1 << i);
            }
        }
        // if (ans > 0x7FFFFFFF) {
        //     ans = ~(ans ^ 0xFFFFFFFF);
        // }
        return ans;
    }
}
